src/main/java/persistence/entity/impl/EntityManagerFactory.java

```diff
@@ -22,31 +25,46 @@
  */
 public class EntityManagerFactory {
 
+    private final SessionContext sessionContext;
     private final Connection connection;
     private final EntityMetaRegistry entityMetaRegistry;
 
     public EntityManagerFactory(Connection connection, EntityMetaRegistry entityMetaRegistry) {
         this.connection = connection;
         this.entityMetaRegistry = entityMetaRegistry;
+        this.sessionContext = initSessionContext(connection, entityMetaRegistry);
     }
 
-    public EntityManager createEntityManager() {
+    private SessionContext initSessionContext(Connection connection, EntityMetaRegistry entityMetaRegistry) {
+        return ThreadLocalSessionContext.init(this, entityManagerFactory ->
+            buildEntityManager(connection, entityMetaRegistry)
+        );
+    }
+
+    public EntityManager openSession() {
+        final EntityManager entityManager = sessionContext.getEntityManager(this);
+        if (entityManager != null) {
+            return entityManager;
+        }
+
+        return sessionContext.bindEntityManager(this, entityManagerFactory ->
+            buildEntityManager(connection, entityMetaRegistry)
+        );
```

Comment:
```suggestion
        return sessionContext.getEntityManager(this);;
```

_해당 코멘트는 제가 구현하신 전체적인 구조를 잘몰라서 그럴수도 있어요!_

저는 위처럼 적용해볼것 같아요.
왜냐하면, 어차피 `sessionContext`는 `EntityManager`를 알고있고 
`getEntityManager`호출 할때 `EntityManager`없으면 `Map`에 넣고 바로 리턴하는게 더 좋지 않을까 생각되긴해요.

Reply:
피드백 주신 방향도 고려했었는데, getEntityManager()를 하는 곳에서 "암시적"으로 Map에 넣어준다는 부분이 설계방향과 맞지 않다고 생각했었는데 차라리 메서드명을 적절하게 수정하고 SessionContext가 해당 역할을 갖고 가도 되겠네요!